### ISSUE TITLE
Upload to swim.com

### ISSUE THREAD

**Issue by ihewitt** [ihewitt/poolviewer#10]
Add upload to swim.com function. XML format looks straightforward to generate.

**Comment by audetto** [comment]
I particularly like the bit where the user id & password are sent in cleartext in the xml!

**Comment by audetto** [comment]
I've tried to use the Windows app to upload to swim.com.
I've edited the data using poolviewer, then imported to windows.

The windows app crashes when it tried to upload to swim.com. I've found out that if I export and reimport the data it will work.

It seems the windows app does some clean up, which i think is related to fixing the best, min, average per set. Which seems strange as a reason for a crash.

I have not looked into it much,

if anyone has time I put them here
before http://pastebin.com/TGEUC7fN
after http://pastebin.com/4PTSMuJh

**Comment by ihewitt** [comment]
yes the windows app crashes if it doesn't like the csv data. 
I try to keep compatibility mostly but I'm not bothering too hard now that 
the linux app is more functional.

On Friday 30 Oct 2015 13:33:08 Andrea wrote:
> I've tried to use the Windows app to upload to swim.com.
> I've edited the data using poolviewer, then imported to windows.
> 
> The windows app crashes when it tried to upload to swim.com. I've 
found out
> that if I export and reimport the data it will work.
> 
> It seems the windows app does some clean up, which i think is 
related to
> fixing the best, min, average per set. Which seems strange as a 
reason for
> a crash.
> 
> I have not looked into it much,
> 
> if anyone has time I put them here
> before http://pastebin.com/TGEUC7fN
> after http://pastebin.com/4PTSMuJh
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> 
https://github.com/ihewitt/poolviewer/issues/10#issuecomment-152644
756

**Comment by audetto** [comment]
Yes,
I've only used the windows app today to try swim.com

**Comment by audetto** [comment]
I was trying to see if there is any other command available

e.g. check if a workout is already there..., remove a workout...

the windows app keeps a copy of all the xml uploaded and I guess it uses this info to detect if something is already there.
I tried to change a workout (merge 2 laps) and ask it to upload all the new ones, but it did not upload it again.

I tried to delete one of these .xml files and ask it to reupload everything, it did try to reupload this workout, but swim.com replied with an error

{"message":"This workout has already been uploaded: Morning Swim","result":"","statusCode":100}

Don't know if one can upload everything all the times and rely on swim.com to ignore the duplicates.
